### PR TITLE
Expose set_weight_decay_lambda() to Python

### LINKED ISSUE #1093: Allow changing weight decay after initialization
Currently weight decay is set once before DyNet is initialized and can't be changed during the whole runtime, even if training a new model.
It should be possible to define a new weight decay for a new model. This is essential for hyperparameter tuning, for example.
Related to #1071.

### PR DESCRIPTION
**Pull request by msperber**
This fixes #1093

### CODE REVIEW

**Comment by neubig** [comment]
Thanks!

**Comment by danielhers** [comment]
Thanks @msperber. What's the difference between `set_weight_decay_lambda()` and `set_weight_decay()`?

**Comment by danielhers** [comment]
Got it, `set_weight_decay()` is a member of `DynetParams` while `set_weight_decay_lambda()` is a member of `ParameterCollection`.

**Comment by talbaumel** [comment]
Any chance you might expose `set_random_seed`?

**Comment by neubig** [comment]
@talbaumel Could you make a new issue?
